lib/sdn_epc/supervisor.ex

```diff
@@ -0,0 +1,28 @@
+defmodule SdnEpc.Supervisor do
+  use Supervisor
+
+  def start_link do
+    res = Supervisor.start_link __MODULE__, [], name: __MODULE__
+    start_ofp_channel_sup()
+    start_worker()
+    res
+  end
+
+  def start_ofp_channel_sup do
+    ofpc_sup_spec = supervisor(:ofp_channel_sup, [1])
+    {:ok, ofpc_sup } = Supervisor.start_child __MODULE__, ofpc_sup_spec
+    Process.register ofpc_sup, :ofp_channel_sup
```

Comment:
Because you will have one channel supervisor for each switch you cannot register them - only one process can be registered under a certain name.